src/ngraph/builder/reshape.cpp

```diff
@@ -177,10 +178,28 @@ shared_ptr<Node> builder::expand_dims(const Output<Node>& value, size_t axis)
         ->add_provenance_group_members_above({value});
 }
 
+NGRAPH_API
 shared_ptr<Node> builder::opset1::reshape(const Output<Node>& value, const Shape& shape)
 {
-    const auto out_pattern = op::Constant::create(
-        element::i64, Shape{shape.size()}, vector<int64_t>(shape.begin(), shape.end()));
+    shared_ptr<op::Constant> out_pattern;
```

Comment:
Could you move `out_pattern` down into the `else` and move `special_zero` and the `return` into the `else` body? Not sure why `special_zero` needs a variable unless it's documentation; could be a constexpr then.

Reply:
Done.